src/Carnac/CarnacTrayIcon.cs

```diff
@@ -39,7 +41,16 @@ public CarnacTrayIcon()
         void NotifyIconClick(object sender, MouseEventArgs mouseEventArgs)
         {
             if (mouseEventArgs.Button == MouseButtons.Left)
-                OpenPreferences();
+                if (Application.Current.Windows.Cast<Window>().Any(x=>x.Name=="PreferencesViewWindow"))
+                {
+                    Application.Current.Windows.Cast<Window>()
```

Comment:
We're duplicating the `Application.Current.Windows.Cast<Window>()` setup and then running other LINQ operations against it. Could we maybe do this?

```c#
var preferenceWindow = Application.Current.Windows.Cast<Window>().FirstOrDefault(x => x.Name == "PreferencesViewWindow");
if (preferenceWindow != null)
{
  preferenceWindow.Activate();
}
else 
{
  OpenPreferences();
}
```